### ISSUE TITLE
Reduce action creator boilerplate

### ISSUE THREAD

**Issue by dansjensen** [shakacode/reactrails-react-native-client#4]
Should we introduce an utils to reduce boilerplate for action creators?

```
function makeActionCreator(type, ...argNames) {
  return function(...args) {
    let action = { type }
    argNames.forEach((arg, index) => {
      action[argNames[index]] = args[index]
    })
    return action
  }
}
```

Source: http://redux.js.org/docs/recipes/ReducingBoilerplate.html

**Comment by justin808** [comment]
IMHO, definitely worth considering.

> There are also utility libraries to aid in generating action creators, such as redux-act and redux-actions. These can help reduce boilerplate code and enforce adherence to standards such as Flux Standard Action (FSA).

CC: @robwise @alexfedoseev

**Comment by robwise** [comment]
I agree, I've always wanted to try this. You could also check out some of the libraries out there: 
- https://github.com/pauldijou/redux-act
- https://github.com/acdlite/redux-actions

There is also a standard called [Flux Standard Action (FSA)](https://github.com/acdlite/flux-standard-action)
